src/main/java/wolox/training/models/Book.java

```diff
@@ -49,79 +50,122 @@ public class Book {
     public long getId() {
         return id;
     }
-    public void setId(long id){
-        this.id = id;
-    }
+
     public String getGenre() {
         return genre;
     }
 
     public void setGenre(String genre) {
-        this.genre = genre;
+        try{
+            Preconditions.checkNotNull(genre, "Illegal Argument, parameter is null");
+            this.genre = genre;
+        }catch(NullPointerException e){
```

Comment:
If you catch the exception you're shadowing the validation 😅 
The correct thing is to throw the exception 😄

Reply:
same goes for the rest of the validations :)